### PR TITLE
Makefile fixes and other spring cleanup

### PR DESCRIPTION
**Pull request by apostasie**
First commit removes DOCKER_BUILDKIT references, which have been the default for a while now.

Second commit focuses on the Makefile:
- most importantly, make it possible to run make targets outside of source tree - clarifying and fixing use of CURDIR vs. newly introduced MAKEFILE_DIR
- introducing `lint` tasks (golint & yamllint) - and also introducing the use of shellcheck
- remove references to GO111MODULE which are no longer useful in 2024
- some minor maintenance (variable-ize calls to `docker`, etc)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,63 +57,76 @@ help:
...
 	@echo " * 'binaries' - Build nerdctl."
 	@echo " * 'clean' - Clean artifacts."
+	@echo " * 'lint' - Run various linters."
 
 nerdctl:
-	$(GO_BUILD) $(GO_BUILD_FLAGS) $(VERBOSE_FLAG) -o $(CURDIR)/_output/nerdctl$(BIN_EXT) $(PACKAGE)/cmd/nerdctl
+	$(GO_BUILD) $(GO_BUILD_FLAGS) $(VERBOSE_FLAG) -o $(CURDIR)/_output/nerdctl$(BIN_EXT) ./cmd/nerdctl
 
 clean:
 	find . -name \*~ -delete
```
Comment (apostasie) [inline Makefile:66]: I am not sure what or why these two are for and if we do need these.
Is that to cleanup leftover files from vi?

Let me know if this should be moved to `find $(MAKEFILE_DIR)`.

File: cmd/nerdctl/builder_build_test.go [hunk 2]
```diff
@@ -45,7 +45,6 @@ CMD ["echo", "nerdctl-build-test-string"]
 	base.Cmd("build", buildCtx, "-t", imageName).AssertOK()
 	base.Cmd("run", "--rm", imageName).AssertOutExactly("nerdctl-build-test-string\n")
 
-	// DOCKER_BUILDKIT (v20.10): `Error response from daemon: exporter "docker" could not be found`
 	if base.Target == testutil.Nerdctl {
```
Comment (AkihiroSuda) [inline cmd/nerdctl/builder_build_test.go:48]: The comment above should be kept, unless we can remove this base.Target check too

Comment (apostasie) [reply cmd/nerdctl/builder_build_test.go:48]: Fair point.
Let me see about that.

Comment (apostasie) [reply cmd/nerdctl/builder_build_test.go]: Base target check removed on latest push (works locally with docker 27 - pending CI).

**Comment by apostasie** [comment]
@AkihiroSuda comment addressed (base check removed) - CI is green

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**